csvGenerator.ipynb

```diff
@@ -0,0 +1,195 @@
+    "            ids.add((values[\"extractedText\"][\"id\"],values[\"extractedText\"][\"filename\"]))\n",
+    "        if values[\"pages\"] and values[\"pages\"] != \"None\":\n",
+    "            for value in values[\"pages\"]:\n",
+    "                ids.add((value[\"image\"][\"id\"],value[\"image\"][\"filename\"]))\n",
+    "                ids.add((value[\"text\"][\"id\"],value[\"text\"][\"filename\"]))\n",
+    "                ids.add((value[\"ocr\"][\"id\"],value[\"ocr\"][\"filename\"]))\n",
+    "                for file in value[\"additionalFiles\"]:\n",
+    "                    ids.add((file[\"id\"],file[\"filename\"]))\n",
+    "    return ids\n",
+    "   \n",
+    "#Create the CSV file with all the metadata and file path of the downloaded files.\n",
+    "\n",
+    "\"\"\"\n",
+    "this function writes the meta data to the csv file.\n",
+    "uses a flag to write the fileds at the top of the file in the first time.\n",
+    "\"\"\"\n",
+    "\n",
+    "def write_to_csv(csv_name:str,item:list,flag:int):\n",
+    "    with open(csv_name, 'a', newline='') as file:\n",
+    "        writer = csv.writer(file)\n",
+    "        if flag==0:\n",
+    "            fields = list(item.keys())\n",
+    "            writer.writerow(list(fields))\n",
+    "        writer.writerow(list(item.values()))\n",
+    "        \n",
+    "# Add the item to csv_dict that contains all the collection items(with metadata and downloaded file path)\n",
+    "\n",
+    "\"\"\"\n",
+    "adds items to a dictionary object; csv_dict \n",
+    "with the collection items unique key as the key for the dictionary\n",
+    "and the item dictionary as the value for the respective key, \n",
+    "that contains our fields and value/metadata for a\n",
+    "given collection item that we write to the csv file. \n",
+    "It also adds a path property that indicates the location of the downloades files(if any)\n",
+    "\"\"\"\n",
+    "\n",
+    "def add_to_csv(csv_name:str,folder_name:str, items:list,csv_dict:dict,flag:int):\n",
+    "    for item in items[\"items\"]:\n",
+    "        if item[\"key\"] in csv_dict:\n",
+    "            continue\n",
+    "        item[\"path\"] = \"\"\n",
+    "        if item[\"gilesUploads\"]:\n",
+    "            giles_ids = get_set(item[\"gilesUploads\"])\n",
+    "            if giles_ids:\n",
+    "                item[\"path\"] = download_file(folder_name, giles_ids, auth_object.access_token)\n",
+    "        csv_dict[item[\"key\"]] = item\n",
+    "        write_to_csv(csv_name,item,flag)\n",
+    "        flag = 1\n",
+    "    return flag"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "229c7e47",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "#downloads and generate a csv file containing all the group items information\n",
+    "def process_groups(csv_name:str,folder_path:str,groups:list, connector, max_size:int):\n",
```

Comment:
what is `groups`? what should it contain, what's it's format, why is it here?

Reply:
documented groups